blockdag/reachability.go

```diff
@@ -227,23 +283,17 @@ func (rtn *reachabilityTreeNode) addChild(child *reachabilityTreeNode) ([]*reach
 	}
 
 	// Allocate from the remaining space
-	allocated, remaining, err := rtn.remainingInterval.splitInHalf()
+	allocated, _, err := remaining.splitInHalf()
 	if err != nil {
 		return nil, err
 	}
 	child.setInterval(allocated)
-	rtn.remainingInterval = remaining
-	return []*reachabilityTreeNode{rtn, child}, nil
+	return map[*reachabilityTreeNode]struct{}{rtn: {}, child: {}}, nil
 }
 
 // setInterval sets the reachability interval for this node.
 func (rtn *reachabilityTreeNode) setInterval(interval *reachabilityInterval) {
```

Comment:
The code sometimes uses setInterval and sometimes sets interval directly. Since setInterval no longer sets the 'remaining' field, we should either use it all the time or delete and not use it at all